src/index/hnsw.cpp

```diff
@@ -378,7 +383,7 @@ HNSW::range_search(const DatasetPtr& query,
             ->NumElements(1)
             ->Ids(ids)
             ->Distances(dists)
-            ->Owner(true, allocator_);
+            ->Owner(true, allocator_.get());
```

Comment:
Here need to passing the shared_ptr<Allocator>, the `Result/Allocator` have different life-cycle

Reply:
I have modified this part to not pass in SafeAllocator, but rather to directly use the raw allocator. This allocator is managed by the caller, which avoids inconsistencies in life-cycle.